diff --git a/MaciASL/DocumentController.m b/MaciASL/DocumentController.m
index b0c5c8a..2fa70d7 100644
--- a/MaciASL/DocumentController.m
+++ b/MaciASL/DocumentController.m
@@ -122,7 +122,7 @@
             file = save.URL.path;
     }
     if (file && [NSFileManager.defaultManager fileExistsAtPath:file] && [[NSFileManager.defaultManager contentsAtPath:file] isEqualToData:aml])
-        [self openDocumentWithContentsOfURL:[NSURL fileURLWithPath:file] display:true completionHandler:nil];
+        [self openDocumentWithContentsOfURL:[NSURL fileURLWithPath:file] display:true completionHandler:^(NSDocument* document, BOOL documentWasAlreadyOpen, NSError *error){}];
     else {
         NSDictionary *decompile = [iASL decompile:aml withResolution:kSystemTableset];
         if ([[decompile objectForKey:@"status"] boolValue])
