diff --git a/NightShiftUnlocker/kern_start.cpp b/NightShiftUnlocker/kern_start.cpp
index 87db307..e8b9ca2 100644
--- a/NightShiftUnlocker/kern_start.cpp
+++ b/NightShiftUnlocker/kern_start.cpp
@@ -12,6 +12,9 @@
 #include <Headers/kern_mach.hpp>
 #include <sys/vnode.h>
 
+#undef memcpy
+#define memcpy lmemcpy
+
 /**
     Read bytes from the given path
  
@@ -195,7 +198,7 @@ static void generatePatches(void* unused, KernelPatcher &patcher)
         return;
     }
     
-    IOLog("NightShiftUnlocker::generatePatches() Found symbol @ %02X.\n", address);
+    IOLog("NightShiftUnlocker::generatePatches() Found symbol @ %02llX.\n", address);
     
     size_t length = numBytesBufferAA;
     
@@ -203,7 +206,7 @@ static void generatePatches(void* unused, KernelPatcher &patcher)
     
     if (buffer != NULL)
     {
-        IOLog("NightShiftUnlocker::generatePatches() %d bytes have been read.\n", length);
+        IOLog("NightShiftUnlocker::generatePatches() %zu bytes have been read.\n", length);
         
         // TODO: Future: Multiple Patches???
         size_t offset = numBytesPatchAA;
