diff --git a/ffsengine.cpp b/ffsengine.cpp
index 0747e90..eb6eaa9 100644
--- a/ffsengine.cpp
+++ b/ffsengine.cpp
@@ -2662,6 +2662,14 @@ UINT8 FfsEngine::rebuild(const QModelIndex & index)
     return ERR_SUCCESS;
 }
 
+UINT8 FfsEngine::doNotRebuild(const QModelIndex & index)
+{
+    if (!index.isValid())
+        return ERR_INVALID_PARAMETER;
+
+    return ERR_SUCCESS;
+}
+
 // Compression routines
 UINT8 FfsEngine::decompress(const QByteArray & compressedData, const UINT8 compressionType, QByteArray & decompressedData, UINT8 * algorithm)
 {
