diff --git a/GPUSensors/RadeonSensors/atom.cpp b/GPUSensors/RadeonSensors/atom.cpp
index 65cdd370..c89f6d0a 100644
--- a/GPUSensors/RadeonSensors/atom.cpp
+++ b/GPUSensors/RadeonSensors/atom.cpp
@@ -48,12 +48,12 @@ static inline uint16_t get_u16(void *bios, int ptr)
 }
 #define U16(ptr) get_u16(ctx->ctx->bios, (ptr))
 #define CU16(ptr) get_u16(ctx->bios, (ptr))
-static inline uint32_t get_u32(void *bios, int ptr)
+/*static inline uint32_t get_u32(void *bios, int ptr)
 {
     return get_u16(bios, ptr)|(((uint32_t)get_u16(bios, ptr+2))<<16);
 }
 #define U32(ptr) get_u32(ctx->ctx->bios, (ptr))
-#define CU32(ptr) get_u32(ctx->bios, (ptr))
+#define CU32(ptr) get_u32(ctx->bios, (ptr))*/
 #define CSTR(ptr) (((char *)(ctx->bios))+(ptr))
 
 static int atom_iio_len[] = { 1, 2, 3, 3, 3, 3, 4, 4, 4, 3 };
