diff --git a/IntelMausiEthernet/IntelMausiEthernet.cpp b/IntelMausiEthernet/IntelMausiEthernet.cpp
index 32e0604..7453fec 100644
--- a/IntelMausiEthernet/IntelMausiEthernet.cpp
+++ b/IntelMausiEthernet/IntelMausiEthernet.cpp
@@ -21,6 +21,10 @@
 
 #include "IntelMausiEthernet.h"
 
+#if __clang_major__ >= 4
+    #pragma clang diagnostic ignored "-Waddress-of-packed-member"
+#endif
+
 #pragma mark --- function prototypes ---
 
 static inline void prepareTSO4(mbuf_t m, UInt32 *mssHeaderSize, UInt32 *payloadSize);
@@ -1970,10 +1974,9 @@ bool IntelMausi::checkForDeadlock()
         if (++deadlockWarn >= kTxDeadlockTreshhold) {
             mbuf_t m = txBufArray[txDirtyIndex].mbuf;
             UInt32 pktSize;
-            UInt16 index;
             
 #ifdef DEBUG
-            UInt16 i;
+            UInt16 i, index;
             UInt16 stalledIndex = txDirtyIndex;
 #endif
             //UInt8 data;
