diff --git a/Multitouch Support/CSGesture/VoodooI2CCSGestureEngine.cpp b/Multitouch Support/CSGesture/VoodooI2CCSGestureEngine.cpp
index 0b13a05..24228ba 100644
--- a/Multitouch Support/CSGesture/VoodooI2CCSGestureEngine.cpp	
+++ b/Multitouch Support/CSGesture/VoodooI2CCSGestureEngine.cpp	
@@ -887,11 +887,11 @@ bool VoodooI2CCSGestureEngine::start(IOService *service) {
     uint16_t max_x = interface->logical_max_x;
     uint16_t max_y = interface->logical_max_y;
     
-    uint16_t hw_res_x = 401;
-    uint16_t hw_res_y = 262;
+    //uint16_t hw_res_x = 401;
+    //uint16_t hw_res_y = 262;
     
-    sprintf(softc.product_id, "ELAN");
-    sprintf(softc.firmware_version, "0561");
+    snprintf(softc.product_id, 5, "ELAN");
+    snprintf(softc.firmware_version, 5, "0561");
     
     softc.resx = max_x;
     softc.resy = max_y;
diff --git a/VoodooI2C/VoodooI2C/VoodooI2CController/VoodooI2CControllerDriver.cpp b/VoodooI2C/VoodooI2C/VoodooI2CController/VoodooI2CControllerDriver.cpp
index 9a84c73..89dd2bd 100644
--- a/VoodooI2C/VoodooI2C/VoodooI2CController/VoodooI2CControllerDriver.cpp
+++ b/VoodooI2C/VoodooI2C/VoodooI2CController/VoodooI2CControllerDriver.cpp
@@ -160,7 +160,7 @@ IOReturn VoodooI2CControllerDriver::prepareTransferI2C(VoodooI2CControllerBusMes
 
     nanoseconds_to_absolutetime(10000, &abstime);
 
-    sleep = command_gate->commandSleep(&bus_device->command_complete, abstime);
+    sleep = command_gate->commandSleep(&bus_device->command_complete, (UInt32)abstime);
 
     if ( sleep == THREAD_TIMED_OUT ) {
         IOLog("%s::%s Timeout waiting for bus to accept transfer request\n", getName(), bus_device->name);
